packages/wrangler/src/index.tsx

```diff
@@ -960,6 +965,49 @@ function createCLIParser(argv: string[]) {
 
 						const accountId = await requireAuth(config);
 
+						if (args.json) {
+							logger.log(
+								`🌀 Creating the secret for the Worker "${scriptName}" ${
+									args.env && !isLegacyEnv(config) ? `(${args.env})` : ""
+								}`
+							);
+							const jsonFilePath = path.resolve(args.json);
+							const content = parseJSON<Record<string, string>>(
+								readFileSync(jsonFilePath),
+								jsonFilePath
+							);
+							for (const key in content) {
+								if (typeof content[key] !== "string") {
+									throw new Error(
+										`The value for ${key} in ${jsonFilePath} is not a string.`
+									);
+								}
+							}
+
+							const url =
+								!args.env || isLegacyEnv(config)
+									? `/accounts/${accountId}/workers/scripts/${scriptName}/secrets`
+									: `/accounts/${accountId}/workers/services/${scriptName}/environments/${args.env}/secrets`;
+							// Until we have a bulk route for secrets, we need to make a request for each key/value pair
+							for (const [key, value] of Object.entries(content)) {
+								await fetchResult(url, {
+									method: "PUT",
+									headers: { "Content-Type": "application/json" },
+									body: JSON.stringify({
+										name: key,
+										text: value,
+										type: "secret_text",
+									}),
+								}).catch((e) => {
+									throw new Error(
+										`Error uploading secret for key: ${key}: ` + e.message
+									);
+								});
+								logger.log(`✨ Successfully created secret for key: ${key}`);
```

Comment:
As we have to do many individual requests, can we `promise.all()` this so we don't have to wait for them all to happen one after another?

Reply:
Also - is there a rate limit for the API? Can someone get 429'd trying to add 100 secrets?

Reply:
Good points. Let's see if we can find/create an API endpoint that can take multiple secrets...

Reply:
@JacobMGEvans in that case, I'd probably make this a `Promise.allSettled` and send back a list of secrets we failed to set

Reply:
> @JacobMGEvans in that case, I'd probably make this a `Promise.allSettled` and send back a list of secrets we failed to set

I didn't even see this and ended up using `.allSettled` lol great minds 😆